### PR TITLE
Authorization: Custom role/permission guards

### PR DESCRIPTION
**Pull request by cherylli**
# Description

Added roles and permissions custom guards and decorators. More roles and permissions can be added in future

As discussed on discord,

1. I don't think @Roles(Voyager) is really needed as OWN_TEAM permission guard works with restricting users to only access their own voyage regardless of their roles (which also works with previous voyages, even if they don't have the voyager role anymore, since voyager role = active voyage participant). But I think there's still value in keeping the voyager role as we can pull everyone with "voyager" role from the database easily

2. User role is probably redundant as as every registered user would be a "USER"

Manual seeding will be required after approval and successful database migration

We will need to update other routes/controllers when this is approved. Also remove `validateLoggedInAndTeamMember` in `global.service`. 

I have updated `users/teams` as an example of how this works

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally on swagger

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: prisma/seed/users.ts
Comment (curtwl) [inline prisma/seed/users.ts:1]: I got "Property 'userRole' does not exist on type 'PrismaClient<PrismaClientOptions, never, DefaultArgs>'." here...looks like it was renamed to VoyageRole?

Comment (cherylli) [reply prisma/seed/users.ts:1]: yes, not sure why it's not updated in the code! 
This is what I have locally `import VoyageRoles from "./data/voyage-roles";`

Comment (cherylli) [reply prisma/seed/users.ts:1]: actually, there are 2 roles now, roles and voyage-roles, did you run `prisma db push`?
roles is the newly added table for "chingu roles", the existing roles are "voyage roles"

Comment (cherylli) [reply prisma/seed/users.ts:1]: It looks like prisma.schema is not committed, will check later

**Comment by cherylli** [comment]
It should be working again @curtwl, thanks for catching that. Please let me know if there's any other issues

**Review by jenny-alexander — APPROVED** [review]
LGTM

**Review by curtwl — APPROVED** [review]
Looks good!

**Pull request merged**